core/src/main/java/org/sonarsource/sonarlint/core/websocket/WebSocketService.java

```diff
@@ -269,7 +289,9 @@ private void resubscribeAll() {
   private void createConnectionIfNeeded(String connectionId) {
     connectionIdsInterestedInNotifications.add(connectionId);
     if (this.sonarCloudWebSocket == null) {
-      this.sonarCloudWebSocket = SonarCloudWebSocket.create(connectionAwareHttpClientProvider.getHttpClient(connectionId), eventRouter::handle, this::reopenConnectionOnClose);
+      this.sonarCloudWebSocket = SonarCloudWebSocket.create(connectionAwareHttpClientProvider.getHttpClient(connectionId),
+        event -> this.eventRouterByConnectionId.get(connectionId).handle(event),
```

Comment:
Here I think it's better to null-check the value returned from the map, in case we receive an event after the connection is closed. It could happen if we just closed the connection but an event arrived at the same time, but I reckon it's quite unlikely